### PR TITLE
Story: [CCLS 2191] Use common auth starter

### PR DESCRIPTION
**Pull request by farrell-m**
* Use [common auth starter](https://github.com/ministryofjustice/laa-ccms-spring-boot-common/pull/4)
  * Added application properties to configure auth
* Added 403 Forbidden to endpoints in OpenAPI spec
* Added configuration to handle authentication during testing

**DRAFT**: Common library version needs to be removed (to use version provided by SB plugin) when Auth Starter PR is merged.

### CODE REVIEW

**Comment by PhilDigitalJustice** [comment]
Can we just update the readme, stating it uses the common library

File: data-service/src/integrationTest/java/uk/gov/laa/ccms/data/IntegrationTestInterface.java [hunk 1]
```diff
@@ -19,6 +19,11 @@ static void properties(DynamicPropertyRegistry registry) {
     registry.add("spring.datasource.url", oracleContainerSingleton.getOracleContainer()::getJdbcUrl);
     registry.add("spring.datasource.username", oracleContainerSingleton.getOracleContainer()::getUsername);
     registry.add("spring.datasource.password", oracleContainerSingleton.getOracleContainer()::getPassword);
+
+    registry.add("laa.ccms.springboot.starter.auth.authentication-header", () -> "Authorization");
```
Comment (porritta) [inline data-service/src/integrationTest/java/uk/gov/laa/ccms/data/IntegrationTestInterface.java:23]: Should this not be in the integrationTest application.yml?

**Review by porritta — CHANGES_REQUESTED** [review]
comments added

File: data-service/src/integrationTest/java/uk/gov/laa/ccms/data/IntegrationTestInterface.java (hunk 1 above)
Comment (farrell-m) [reply data-service/src/integrationTest/java/uk/gov/laa/ccms/data/IntegrationTestInterface.java]: Agree, that makes more sense. Moved.

**Review by PhilDigitalJustice — APPROVED** [review]
LGTM

**Pull request merged**